typedlua/tlchecker.lua

```diff
@@ -1962,9 +1962,14 @@ function tlchecker.typecheck (ast, subject, filename, strict, integer, color)
   return env.messages
 end
 
-local function get_source_line(filename, l)
+local function get_source_line(subject, filename, l)
+  if not subject then
+    local file = io.open(filename)
+    subject = file:read("*a")
+    file:close()
+  end
   local i = 1
-  for source_line in io.lines(filename) do
+  for source_line in subject:gmatch("[^\r\n]*") do
```

Comment:
I think this will result in skipping empty lines and getting the count wrong?

Reply:
In Lua < 5.3.4 (yes it was changed in a minor version!) gmatch will return extra 0 length matches.

See e.g. 
```lua
for thing in ("a\nb\n\nfoo"):gmatch("[^\r\n]*") do print(thing) end
```

Reply:
It was undefined behaviour that finally got changed to the "expected" behaviour in the 5.3.3 release. I believe for 5.4 the behaviour will actually be documented (and hence relied on).

> I guess I will have to write a custom iterator using string.byte...

Just modify your gmatch slightly. `^(.-)\r?\n` should work I think? (excluding the final line of the file).
Though you have lpeg as a dependency already! Make something nicer :)

Reply:
Using lpeg for that is shooting a fly with a bazooka. :-) I just did a loop using `string.find` looking for `\n`.